src/topology/algebra/module/linear_pmap.lean

```diff
@@ -0,0 +1,161 @@
+Authors: Moritz Doll
+-/
+
+import linear_algebra.linear_pmap
+import topology.algebra.module.basic
+
+/-!
+# Linear operators over topological vector spaces
+
+We define basic notions of linear operators (also known as `linear_pmap`s in mathlib).
+In this file we prove all elementary properties that do not assume that the underlying spaces
+are normed.
+
+## Main definitions
+
+* `linear_pmap.closed`: an operator is closed iff it's graph is closed
+* `linear_pmap.closable`: an operator is closable iff the closure of it's graph is a graph
+* `linear_pmap.closure`: the closure of a closable operator
+* `linear_pmap.core`: a submodule contained in the domain is a core if restricting to the core
+  does not lose information about the operator
+
+## Main statements
+
+* `closable_iff_exists_closed_extension`: an operator is closable iff it has a closed extension
+* `closable.exists_unique`: there exists a unique closure
+
+## References
+
+* [J. Weidmann, *Linear Operators in Hilbert Spaces*][weidmann_linear]
+
+## Tags
+
+Unbounded operators, closed operators
+-/
+
+
+open_locale topological_space
+
+variables {R E F 𝕜: Type*}
+
+variables [comm_ring R] [add_comm_group E] [add_comm_group F]
+variables [module R E] [module R F]
+variables [topological_space E] [topological_space F]
+
+namespace linear_pmap
+
+/-! ### Closed and closable operators -/
+
+/-- An operator is closed if its graph is closed. -/
+def closed (f : linear_pmap R E F) : Prop :=
+is_closed (f.graph : set (E × F))
+
+variables [has_continuous_add E] [has_continuous_add F]
+variables [topological_space R] [has_continuous_smul R E] [has_continuous_smul R F]
+
+/-- An operator is closable if the closure of the graph is a graph. -/
+def closable (f : linear_pmap R E F) : Prop :=
+∃ (f' : linear_pmap R E F), f.graph.topological_closure = f'.graph
+
+/-- A closed operator is trivially closable. -/
+lemma closed.closable {f : linear_pmap R E F} (hf : f.closed) : f.closable :=
+⟨f, hf.submodule_topological_closure_eq⟩
+
+/-- If `f` has a closable extension `g`, then `f` itself is closable. -/
+lemma closable.le_closable {f g : linear_pmap R E F} (hf : f.closable) (hfg : g ≤ f) : g.closable :=
+begin
+  cases hf with f' hf,
+  have : g.graph.topological_closure ≤ f'.graph :=
+  by { rw ←hf, exact submodule.topological_closure_mono (le_graph_of_le hfg) },
+  refine ⟨g.graph.topological_closure.to_linear_pmap _, _⟩,
+  { intros x hx hx',
+    cases x,
+    exact f'.graph_fst_eq_zero_snd (this hx) hx' },
+  rw [submodule.to_linear_pmap_graph_eq],
+end
+
+/-- The closure is unique. -/
+lemma closable.exists_unique {f : linear_pmap R E F} (hf : f.closable) :
+  ∃! (f' : linear_pmap R E F), f.graph.topological_closure = f'.graph :=
+begin
+  refine exists_unique_of_exists_of_unique hf (λ _ _ hy₁ hy₂, eq_of_eq_graph _),
+  rw [←hy₁, ←hy₂],
+end
+
+/-- The closure of a closable operator. -/
+noncomputable
+def closable.closure {f : linear_pmap R E F} (hf : f.closable) : linear_pmap R E F :=
```

Comment:
I'd rather see a definition that does not depend on `hf`, with a junk value of `f` is not closable. Something like, if `f` is closable, then use a closure, else use `f`. Then many of the lemmas below would still be true, and this would make closures easier to use.

Reply:
Thank you so much for that suggestion, that really makes things way more readable and removes the stupid `congr` lemma I had to write.